runtime/ftplugin/sh.vim

```diff
@@ -44,19 +44,30 @@ if (has("gui_win32") || has("gui_gtk")) && !exists("b:browsefilter")
   let b:undo_ftplugin ..= " | unlet! b:browsefilter"
 endif
 
-if get(b:, "is_bash", 0)
+let s:is_sh = get(b:, "is_sh", 0)
+let s:is_bash = get(b:, "is_bash", 0)
+let s:is_kornshell = get(b:, "is_kornshell", get(g:, "is_kornsh", 0))
+
+if s:is_bash
   if exists(':terminal') == 2
     command! -buffer -nargs=1 ShKeywordPrg silent exe ':term bash -c "help "<args>" 2>/dev/null || man "<args>""'
   else
     command! -buffer -nargs=1 ShKeywordPrg echo system('bash -c "help <args>" 2>/dev/null || MANPAGER= man "<args>"')
   endif
   setlocal keywordprg=:ShKeywordPrg
   let b:undo_ftplugin ..= " | setl kp< | sil! delc -buffer ShKeywordPrg"
+endif
 
+if executable('shellcheck') && (s:is_sh || s:is_bash || s:is_kornshell)
   if !exists('current_compiler')
     compiler shellcheck
   endif
   let b:undo_ftplugin .= ' | compiler make'
+elseif s:isbash
+  if !exists('current_compiler')
+    compiler bash
+  endif
+  let b:undo_ftplugin .= ' | compiler make'
```

Comment:
```suggestion
  let b:undo_ftplugin ..= ' | compiler make'
```

Shouldn't the restoring of `compiler` be also made conditional  
here and at lines 62-65?

There is also an out-of-style use of `.=` at line 65.